ext/logging.c

```diff
@@ -1,97 +1,171 @@
 #include "logging.h"
+#include "sidecar.h"
 
-#include <stdatomic.h>
+#include <fcntl.h>
 #include <stdio.h>
 #include <string.h>
 #include <time.h>
 
 #include "configuration.h"
 #include <main/SAPI.h>
 
-static inline ddog_CharSlice dd_zend_string_to_CharSlice(zend_string *str) {
-    return (ddog_CharSlice){ .len = str->len, .ptr = str->val };
+
+static void dd_log_set_level(bool debug) {
+    bool once = runtime_config_first_init ? get_DD_TRACE_ONCE_LOGS() : get_global_DD_TRACE_ONCE_LOGS();
+    if (debug) {
+        if (strcmp("cli", sapi_module.name) != 0 && (runtime_config_first_init ? get_DD_TRACE_STARTUP_LOGS() : get_global_DD_TRACE_STARTUP_LOGS())) {
+            ddog_set_log_level(DDOG_CHARSLICE_C("debug"), once);
+        } else {
+            ddog_set_log_level(DDOG_CHARSLICE_C("debug,startup=error"), once);
+        }
+    } else if (runtime_config_first_init) {
+        ddog_set_log_level(dd_zend_string_to_CharSlice(get_DD_TRACE_LOG_LEVEL()), once);
+    } else if (zend_string_equals_literal_ci(Z_STR(zai_config_memoized_entries[DDTRACE_CONFIG_DD_TRACE_LOG_LEVEL].decoded_value), "error")) {
+        ddog_set_error_log_level(once); // optimized handling without parsing
+    } else {
+        ddog_set_log_level(dd_zend_string_to_CharSlice(get_global_DD_TRACE_LOG_LEVEL()), once);
+    }
```

Comment:
So if I understand this correctly, the "least" amount of logs one can set is `error`, right?

Reply:
If DD_TRACE_DEBUG is disabled, error is still present (current state). DD_TRACE_LOG_LEVEL=off would fully disable it.